mhkit/utils/stat_utils.py

```diff
@@ -1,9 +1,57 @@
-from mhkit import qc
+"""
+This module contains functions to perform various statistical calculations 
+on continuous data. It includes functions for calculating statistics such as
+mean, max, min, and standard deviation over specific windows, as well as functions
+for vector/directional statistics. The module also provides utility functions 
+to unwrap vectors, compute magnitudes and phases in 2D/3D, and calculate 
+the root mean squared values of vector components.
+
+Functions:
+----------
+- get_statistics: Calculates statistics for continuous data.
+- vector_statistics: Calculates vector mean and standard deviation.
+- unwrap_vector: Unwraps vector data to fall within a 0-360 degree range.
+- magnitude_phase: Computes magnitude and phase for 2D or 3D data.
+- unorm: Computes root mean squared value of 3D vectors.
+"""
+
 import pandas as pd
 import numpy as np
+from mhkit import qc
+
+
+def _calculate_statistics(datachunk, vector_channels):
+    """
+    Calculate the mean, max, min, and standard deviation for the given datachunk.
+    Also calculate vector statistics for vector_channels.
+
+    Parameters
+    ----------
+    datachunk : pandas DataFrame
+        A chunk of data on which to perform statistics.
+    vector_channels : list
+        List of vector channel names formatted in deg (0-360).
+
+    Returns
+    -------
+    stats : dict
+        A dictionary containing 'means', 'maxs', 'mins', and 'stdevs'.
+    """
+    means = datachunk.mean()
+    maxs = datachunk.max()
+    mins = datachunk.min()
+    stdevs = datachunk.std()
+
+    for v in vector_channels:
+        vector_avg, vector_std = vector_statistics(datachunk[v])
+        # overwrite scalar average and std for channel
+        means[v] = vector_avg
+        stdevs[v] = vector_std
 
+    return {"means": means, "maxs": maxs, "mins": mins, "stdevs": stdevs}
 
-def get_statistics(data, freq, period=600, vector_channels=[]):
+
+def get_statistics(data, freq, period=600, vector_channels=None):
```

Comment:
Removed empty list due to  dangerous default value:
https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/dangerous-default-value.html